test/libsolidity/smtCheckerTests/functions/constructor_hierarchy_diamond.sol

```diff
@@ -0,0 +1,29 @@
+pragma experimental SMTChecker;
+contract C {
+	uint a;
+	constructor(uint x) public {
+		a = x;
+	}
+}
+
+contract B1 is C {
+	constructor(uint x) public {
+		a = x;
+	}
+}
+
+contract B2 is C {
+	constructor(uint x) C(x + 2) public {
+		a = x;
+	}
+}
+
+contract A is B2, B1 {
+	constructor(uint x) B2(x) B1(x) public {
+		assert(a == x);
+		assert(a == x + 1);
+	}
```

Comment:
For these diamond one's it would also be nice to set ``b1 = x + a`` and ``b2 = x + a`` (state variables of ``B1``, resp. ``B2``) instead of ``a`` in the constructors of ``B1`` and ``B2`` and then check those...

Reply:
Added